client/my-sites/posts/post-controls/index.jsx

```diff
@@ -0,0 +1,263 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import url from 'url';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { isEnabled } from 'config';
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+} from 'state/analytics/actions';
+import {
+	canCurrentUser,
+	isPublicizeEnabled,
+} from 'state/selectors';
+import PostControl from './post-control';
+
+export class PostControls extends Component {
+	static propTypes = {
+		canUserDeletePost: PropTypes.bool,
+		canUserEditPost: PropTypes.bool,
+		canUserPublishPost: PropTypes.bool,
+		editURL: PropTypes.string.isRequired,
+		fullWidth: PropTypes.bool,
+		isPublicizeEnabled: PropTypes.bool,
+		onDelete: PropTypes.func,
+		onPublish: PropTypes.func,
+		onRestore: PropTypes.func,
+		onToggleShare: PropTypes.func,
+		onTrash: PropTypes.func,
+		post: PropTypes.object.isRequired,
+		recordCopyPost: PropTypes.func,
+		recordEditPost: PropTypes.func,
+		recordPreviewPost: PropTypes.func,
+		recordViewPost: PropTypes.func,
+		recordViewPostStats: PropTypes.func,
+		site: PropTypes.object,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		showMoreOptions: false,
+	};
+
+	onShowMore = () => this.setState( { showMoreOptions: true } );
+
+	onHideMore = () => this.setState( { showMoreOptions: false } );
+
+	getAvailableControls() {
+		const {
+			canUserDeletePost,
+			canUserEditPost,
+			canUserPublishPost,
+			editURL,
+			fullWidth,
+			onDelete,
+			onPublish,
+			onRestore,
+			onToggleShare,
+			onTrash,
+			post,
+			recordCopyPost,
+			recordEditPost,
+			recordPreviewPost,
+			recordViewPost,
+			recordViewPostStats,
+			site,
+			translate,
+		} = this.props;
+		const controls = { main: [], more: [] };
+
+		// NOTE: Currently Jetpack site posts are not returning `post.capabilities`
+		// and those posts will not have access to post management type controls
+
+		// Main Controls (not behind ... more link)
+		if ( canUserEditPost ) {
+			controls.main.push( {
+				className: 'edit',
+				href: editURL,
+				icon: 'pencil',
+				onClick: recordEditPost,
+				text: translate( 'Edit' ),
+			} );
+		}
+
+		if ( 'publish' === post.status ) {
+			controls.main.push( {
+				className: 'view',
+				href: post.URL,
+				icon: 'external',
+				onClick: recordViewPost,
+				target: '_blank',
+				text: translate( 'View' ),
+			} );
+
+			controls.main.push( {
+				className: 'stats',
+				href: `/stats/post/${ post.ID }/${ site.slug }`,
+				icon: 'stats-alt',
+				onClick: recordViewPostStats,
+				text: translate( 'Stats' ),
+			} );
+
+			if ( isEnabled( 'republicize' ) ) {
+				controls.main.push( {
+					className: 'share',
+					disabled: ! this.props.isPublicizeEnabled,
```

Comment:
@Copons In a dev env, it looks like the share button is broken. For behavior we can compare with wpcalypso.wordpress.com where republicize is also enabled